flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManager.java

```diff
@@ -0,0 +1,381 @@
+
+		numPendingPodRequests++;
+
+		log.info("Requesting new TaskExecutor pod with <{},{}>. Number pending requests {}.",
+			defaultTaskManagerMemoryMB,
+			defaultCpus,
+			numPendingPodRequests);
+
+		final String podName = String.format(
+			TASK_MANAGER_POD_FORMAT,
+			clusterId,
+			currentMaxAttemptId.get(),
+			currentMaxPodId.addAndGet(1));
+
+		try {
+			final HashMap<String, String> env = new HashMap<>();
+			env.put(Constants.ENV_FLINK_POD_NAME, podName);
+
+			final TaskManagerPodParameter parameter = new TaskManagerPodParameter(
+				podName,
+				getTaskManagerStartCommand(podName),
+				defaultTaskManagerMemoryMB,
+				defaultCpus,
+				env);
+
+			this.kubeClient.createTaskManagerPod(parameter);
+		} catch (Exception e) {
+			this.kubeClient.handleException(e);
+			throw new FlinkRuntimeException("Could not start new worker");
+		}
+	}
+
+	/**
+	 * Request new pod if pending pods cannot satisfies pending slot requests.
+	 */
+	private void requestKubernetesPodIfRequired() {
+		final int requiredTaskManagerSlots = getNumberRequiredTaskManagerSlots();
+		final int pendingTaskManagerSlots = numPendingPodRequests * numberOfTaskSlots;
+
+		if (requiredTaskManagerSlots > pendingTaskManagerSlots) {
+			requestKubernetesPod();
+		}
+	}
+
+	private void removePodIfTerminated(KubernetesPod pod) {
+		if (pod.isTerminated()) {
+			kubeClient.stopPod(pod.getName());
+			final KubernetesWorkerNode kubernetesWorkerNode = workerNodeMap.remove(new ResourceID(pod.getName()));
+			if (kubernetesWorkerNode != null) {
+				requestKubernetesPodIfRequired();
+			}
+		}
+	}
+
+	private List<String> getTaskManagerStartCommand(String podName) {
+		final ContaineredTaskManagerParameters taskManagerParameters =
+			ContaineredTaskManagerParameters.create(flinkConfig, taskExecutorResourceSpec, numberOfTaskSlots);
+
+		log.info("TaskExecutor {} will be started with {}.", podName, taskExecutorResourceSpec);
+
+		final String confDir = flinkConfig.getString(KubernetesConfigOptions.FLINK_CONF_DIR);
+		final boolean hasLogback = new File(confDir, Constants.CONFIG_FILE_LOGBACK_NAME).exists();
+		final boolean hasLog4j = new File(confDir, Constants.CONFIG_FILE_LOG4J_NAME).exists();
+
+		final String logDir = flinkConfig.getString(KubernetesConfigOptions.FLINK_LOG_DIR);
+
+		final String mainClassArgs = "--" + CommandLineOptions.CONFIG_DIR_OPTION.getLongOpt() + " " +
+			flinkConfig.getString(KubernetesConfigOptions.FLINK_CONF_DIR) + " " +
+			BootstrapTools.getDynamicProperties(flinkClientConfig, flinkConfig);
+
+		final String command = KubernetesUtils.getTaskManagerStartCommand(
+			flinkConfig,
+			taskManagerParameters,
+			confDir,
+			logDir,
+			hasLogback,
+			hasLog4j,
+			KubernetesTaskExecutorRunner.class.getCanonicalName(),
+			mainClassArgs);
+
+		return Arrays.asList("/bin/bash", "-c", command);
+	}
+
+	/**
+	 * Get task manager label for the current flink cluster. They could be used to watching the pods status.
```

Comment:
watching -> watch, also better to add an empty line before below `@return`

Reply:
I will fix it.